### ISSUE TITLE
[HOLD for payment 2023-03-16] [$2000] Web - Emoji - There is no tooltip for the category picker

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#15108]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Issue found when executing PR** https://github.com/Expensify/App/pull/14532

## Action Performed:
1. Go to staging.new.expensify.com
2. Log in with any account
3. Go to any conversation
4. Select emoji option
5. Hovering on one of the option

## Expected Result:
When hovering over one of the category pickers, a tooltip containing the name of the category should appear. 

## Actual Result:
There is no tooltip for the category picker

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.71.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** No - New feature

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

![image](https://user-images.githubusercontent.com/93399543/218580537-2001229f-a750-4cbb-9a9a-c69335b18f06.png)

https://user-images.githubusercontent.com/93399543/218580554-3dd53e0e-8b4e-48c3-9b52-ac48e84fe8d3.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~010a0b351e969622b9</li>
        <li>Upwork Job ID: 1625929024649732096</li>
        <li>Last Price Increase: 2023-02-22</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @trjExpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by trjExpensify** [comment]
This is a recent feature we released [here](https://github.com/Expensify/Expensify/issues/239100), asking in [this thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1676479338395609) on the expected behaviour.

**Comment by JmillsExpensify** [comment]
Great catch, I agree with Applause that we just overlooked this and should close the gap.

**Comment by trjExpensify** [comment]
Alrighty, moving it on. CC: @stitesExpensify for vis

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~010a0b351e969622b9

**Comment by MelvinBot** [comment]
Current assignee @trjExpensify is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @aimane-chnaif (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @robertjchen (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by PrashantMangukiya** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Within web we have to show tooltip when mouse over on emoji picker category icon.

### What is the root cause of that problem?
Tooltip not used within `<CategoryShortcutButton>` so it will not show tooltip whenever mouse over the cateogry icon.

### What changes do you think we should make in order to solve the problem?
_I think solution is pretty simple and clean as explained below:_

Within `src/components/EmojiPicker/CategoryShortcutBar.js` file we already have an array of category icons as shown below. 

https://github.com/Expensify/App/blob/1180571bf3059d0388f57d1c661ff61f7fc4758a/src/components/EmojiPicker/CategoryShortcutBar.js#L26-L31

So within `src/components/EmojiPicker/CategoryShortcutBar.js` file we have to create 1) Another array that consist translation keys corresponding to category icons and 2) Add translation key for "Frequently used" section if exist and 3) pass new prop named tooltipText to `<CategoryShortcutButton>` component as shown code below:

```javascript
const CategoryShortcutBar = (props) => {
     const icons = [Smiley, AnimalsAndNature, FoodAndDrink, TravelAndPlaces, Activities, Objects, Symbols, Flags];
    
    // *** Add this code ***
    // Prepare array of translation keys to show as tooltip for each category icon
    const tooltipTextKeys = _.map([
        'smileysAndEmotion',
        'animalsAndNature',
        'foodAndDrink',
        'travelAndPlaces',
        'activities',
        'objects',
        'symbols',
        'flags',
    ], item => `emojiPicker.headers.${item}`);

    // If the user has frequently used emojis, there will be 9 headers, otherwise there will be 8
    if (props.headerIndices.length === 9) {
        icons.unshift(FrequentlyUsed);
        tooltipTextKeys.unshift('emojiPicker.headers.frequentlyUsed');    // *** Add this code ***
    }

    return (
        <View style={[styles.pt2, styles.ph4, styles.flexRow]}>
            {_.map(props.headerIndices, (headerIndex, i) => (
                <CategoryShortcutButton
                    icon={icons[i]}
                    onPress={() => props.onPress(headerIndex)}
                    key={`categoryShortcut${i}`}
                    tooltipText={props.translate(tooltipTextKeys[i])}.  // *** Add this code ***
                />
            ))}
        </View>
    );

}
```

Within `src/components/EmojiPicker/CategoryShortcutButton.js` file we have view with icon as shown below:

https://github.com/Expensify/App/blob/1180571bf3059d0388f57d1c661ff61f7fc4758a/src/components/EmojiPicker/CategoryShortcutButton.js#L39-L46

So within `src/components/EmojiPicker/CategoryShortcutButton.js` file we have to define new prop named tooltipText and  wrap View with tooltip as shown code below:
```javascript

const propTypes = {

    /** Tooltip text to show when mouse over (Add this code) */
    tooltipText: PropTypes.string.isRequired,
};

                <Tooltip
                    text={this.props.tooltipText}
                    shiftVertical={-7}
                >
                    <View style={styles.alignSelfCenter}>
                        <Icon
                            fill={themeColors.icon}
                            src={this.props.icon}
                            height={variables.iconSizeNormal}
                            width={variables.iconSizeNormal}
                        />
                    </View>
                </Tooltip>

```
So this solution is efficient and requires minimum code changes.

It will also take care to show tooltip in corresponding language when language changed as shown in result video below. Also it will keep button size consistent in all platform as other platform does not show tooltip.

### What alternative solutions did you explore? (Optional)

None

### Result video:

<details>
<summary>Web</summary>
<br/>

https://user-images.githubusercontent.com/7823358/219662535-ffd73ad4-3a66-48ff-95d1-814d5e6c72ef.mp4
</details>

<details>
<summary>iOS</summary>
<br/>

https://user-images.githubusercontent.com/7823358/219663676-66e8c534-b30e-4837-9662-bc5fcaa767e2.mov
</details>

<details>
<summary>Android</summary>
<br/>

https://user-images.githubusercontent.com/7823358/219662845-7ecd2c3a-0789-49d8-9405-b5275bbd0f19.mov
</details>

**Comment by trjExpensify** [comment]
@aimane-chnaif what do you think of this proposal?

**Comment by JmillsExpensify** [comment]
Assigning myself as well so I can keep track of this one as part of the tracking issue.

**Comment by MelvinBot** [comment]
📣 @daraksha-dk! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by akshayasalvi** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- For the web platforms, we would like to show text tooltips to all the category icons on the Emoji Category Bar

### What is the root cause of that problem?
- I think this is a feature request as when it was designed it seems that only Icons were added. No Tooltip exists as of now for the given component.

### What changes do you think we should make in order to solve the problem?

I think we should update the `CategoryShortcutBar` to accept the `{code , index} []` array of the headers, so that we can apply the tooltip to each of the icon.

1. Add props to `CategoryShortcutBar`, `headerEmojis` which consist of the `code` and `index`.
2. Add a new prop `code` to `CategoryShortcutButton` which takes the headerEmoji.code
3. Add `withLocalize` HoC to `CategoryShortcutButton`, so that we can access the translate method
4. Wrap the following `View`  with `Tooltip`

https://github.com/Expensify/App/blob/a87b802a09c48de7274855af56e27ab544463ef6/src/components/EmojiPicker/CategoryShortcutButton.js#L39

5. For the `Tooltip` use `translate('emojiPicker.headers.${code}')`
6. Update `getHeaderIndices` in EmojiUtils to return the whole object along with the index (or atleast the code and the index)
https://github.com/Expensify/App/blob/a87b802a09c48de7274855af56e27ab544463ef6/src/libs/EmojiUtils.js#L89

7.  In `EmojiPicker/index.js` and can be applied to `native` as well.

https://github.com/Expensify/App/blob/a87b802a09c48de7274855af56e27ab544463ef6/src/components/EmojiPicker/EmojiPickerMenu/index.js#L63

Modify the above block as:

```
this.headerEmojis = = EmojiUtils.getHeaderEmojis(allEmojis);`
this.headerRowIndices = _.map(this.headerEmojis, (headerEmoji) => Math.floor(headerEmoji.index / CONST.EMOJI_NUM_PER_ROW));
```




### What alternative solutions did you explore? (Optional)

I would also recommend one additional refactor. We can move the following imports also to `assets/emoji.js`

https://github.com/Expensify/App/blob/a87b802a09c48de7274855af56e27ab544463ef6/src/components/EmojiPicker/CategoryShortcutBar.js#L7-L15

So that in future whenever we add a new category we only have one place of truth. We don't need to maintain arrays at two places.  We can get rid of `const icons = [Smiley, AnimalsAndNature, FoodAndDrink, TravelAndPlaces, Activities, Objects, Symbols, Flags];` and the code becomes a lot cleaner.

**Comment by MelvinBot** [comment]
📣 @akshayasalvi! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by PrashantMangukiya** [comment]
@aimane-chnaif  here is my proposal 2

## Proposal 2

### Please re-state the problem that we are trying to solve in this issue.

Within web we have to show tooltip when mouse over on emoji picker category icon.

### What is the root cause of that problem?
Tooltip not used within `<CategoryShortcutButton>` so it will not show tooltip whenever mouse over the cateogry icon.

### What changes do you think we should make in order to solve the problem?

Within `src/libs/EmojiUtils.js` file we have to add helper function as shown below that will return header indices with coresponding code.
```javascript
function getHeaderIndicesWithCode(emojis) {
    const headerIndices = [];
    _.each(emojis, (emoji, index) => {
        if (!emoji.header) {
            return;
        }
        headerIndices.push({index, code: emoji.code});
    });
    return headerIndices;
}
```

Within `src/components/EmojiPicker/EmojiPickerMenu/index.js` file we have to update code as shown below:
```javascript

        // This is the actual header index starting at the first emoji and counting each one
        //this.headerIndices = EmojiUtils.getHeaderIndices(allEmojis);    // *** OLD CODE
        const headerIndicesWithCode = EmojiUtils.getHeaderIndicesWithCode(allEmojis);   // *** New Code ****
        this.headerIndices = _.map(headerIndicesWithCode, item => item.index);    // *** New Code ****
        this.headerCodes = _.map(headerIndicesWithCode, item => item.code);        // *** New Code ****

        <CategoryShortcutBar
            headerIndices={this.headerIndices}
            headerCodes={this.headerCodes}.  / / *** Add  this code
            onPress={this.scrollToHeader}
        />
```

So within src/components/EmojiPicker/CategoryShortcutBar.js we have to pass `tooltipText` as props to `CategoryShortcutButton` component.

```javascript
const CategoryShortcutBar = (props) => {

    return (
        <View style={[styles.pt2, styles.ph4, styles.flexRow]}>
            {_.map(props.headerIndices, (headerIndex, i) => (
                <CategoryShortcutButton
                    icon={icons[i]}
                    onPress={() => props.onPress(headerIndex)}
                    key={`categoryShortcut${i}`}
                    tooltipText={props.translate(`emojiPicker.headers.${props.headerCodes[i]}`)} // *** Add this prop ***
                />
            ))}
        </View>
    );
}
```

Within `src/components/EmojiPicker/CategoryShortcutButton.js` file we have view with icon as shown below:

https://github.com/Expensify/App/blob/1180571bf3059d0388f57d1c661ff61f7fc4758a/src/components/EmojiPicker/CategoryShortcutButton.js#L39-L46

So within `src/components/EmojiPicker/CategoryShortcutButton.js` file we have to define new prop named tooltipText and  wrap View with tooltip as shown code below:
```javascript

const propTypes = {

    /** Tooltip text to show when mouse over (Add this code) */
    tooltipText: PropTypes.string.isRequired,
};

                <Tooltip
                    text={this.props.tooltipText}
                    shiftVertical={-7}
                >
                    <View style={styles.alignSelfCenter}>
                        <Icon
                            fill={themeColors.icon}
                            src={this.props.icon}
                            height={variables.iconSizeNormal}
                            width={variables.iconSizeNormal}
                        />
                    </View>
                </Tooltip>

```

### What alternative solutions did you explore? (Optional)

None

**Comment by MelvinBot** [comment]
📣 @PrashantMangukiya! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by MelvinBot** [comment]
📣 @akshayasalvi You have been assigned to this job by @stitesExpensify!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by akshayasalvi** [comment]
@stitesExpensify @aimane-chnaif I can see a PR already for the linked issue so should I wait for the other PR to be merged or can I start with my PR?

**Comment by akshayasalvi** [comment]
Yeah and then I'll have to test if Tooltip behavior doesn't cause issue in the later position of the Category Shortcuts Bar.

I think we should wait. But do let me know if you want me to start I'll start right away (I just don't want this to affect the PR bonus).

**Comment by stitesExpensify** [comment]
Yeah, let's wait until it's merged. I will make sure that the PR bonus is not affected.

**Comment by akshayasalvi** [comment]
@stitesExpensify The PR is now merged, I will now start the PR and raise it by tomorrow.

> I would also recommend one additional refactor. We can move the following imports also to assets/emoji.js

Meanwhile, can you or @aimane-chnaif please confirm if you want this refactor to be done?

**Comment by akshayasalvi** [comment]
@aimane-chnaif @stitesExpensify One more question.

How much displacement should a Tooltip generally have from the icon?

This is how it looks after applying the Tooltip. If you notice it is applied to View, so the hover background is covered by the Tooltip.
<img width="421" alt="Screenshot 2023-02-27 at 11 12 57 PM" src="https://user-images.githubusercontent.com/57435789/221641876-fbfb7b6d-5052-47de-9171-39f4c5600cce.png">


I tried applying the Tooltip to Pressable, but we would have to move the styling to a View and then wrap that View around Tooltip.

<img width="464" alt="Screenshot 2023-02-27 at 11 10 57 PM" src="https://user-images.githubusercontent.com/57435789/221642076-152e8e1f-1305-4706-bf72-adb601218fa3.png">

Lastly, I applied `shiftVertical={-8}`, which I think is apt for this case. I am pushing with this change but happy to update with whatever path you choose.

<img width="498" alt="Screenshot 2023-02-27 at 11 16 10 PM" src="https://user-images.githubusercontent.com/57435789/221642414-cf05e844-6b55-44c1-883c-82226677079e.png">

**Comment by stitesExpensify** [comment]
> please confirm if you want this refactor to be done?

Sure @akshayasalvi let's move those imports to emoji.js

> How much displacement should a Tooltip generally have from the icon?

I'll defer to the expert :) @shawnborton

**Comment by shawnborton** [comment]
I think it should be what you have in your second screenshot, so `shiftVertical={-8}`

**Comment by akshayasalvi** [comment]
Okay @stitesExpensify I'll refactor that and thanks @shawnborton.

**Comment by akshayasalvi** [comment]
PR is raised and ready for review.

**Comment by akshayasalvi** [comment]
@aimane-chnaif Based on this [comment](https://github.com/Expensify/App/issues/15108#issuecomment-1446770718) and this [response](https://github.com/Expensify/App/issues/15108#issuecomment-1446810669), should we go ahead with 2px? And complete the checklist ? 

Or do you recommend waiting more?

**Comment by akshayasalvi** [comment]
@aimane-chnaif I've update the change to have 2px. Because it was somewhat agreed earlier and you're also okay with it. Just to ensure that if I am offline later (and we get confirmation), it shouldn't block you from testing.  If changes are requested, for 0px or 4px, I'll try to do them asap.

Meanwhile, is there any other change/feedback that needs to be worked upon? I am online for a few hours, so I can close them. Thank you very much for your time.

**Comment by akshayasalvi** [comment]
@aimane-chnaif Can you add @stitesExpensify as a reviewer? When I requested your review again I don't know how it removed them. I can't seem to be able to add anybody.

**Comment by shawnborton** [comment]
I think I like 4px since that is a standard spacing unit for us.

**Comment by stitesExpensify** [comment]
For clarity for the bugZero assignee (@trjExpensify) we had this on hold, and then I was OOO and delayed the merging of this. This issue should be eligible for the speed bonus

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.80-2 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15513


If no regressions arise, payment will be issued on 2023-03-16. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@aimane-chnaif / @stitesExpensify] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@aimane-chnaif / @stitesExpensify] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@aimane-chnaif / @stitesExpensify] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@JmillsExpensify / @trjExpensify] Determine if we should create a regression test for the bug.
- [ ] [@aimane-chnaif] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to the [appropriate location](https://github.com/Expensify/App/blob/main/contributingGuides/TestRail_Navigation_C%2B.md) to ensure the same bug will not reach production again.
- [ ] [@JmillsExpensify / @trjExpensify] Review the proposed regression test steps and location.
- [ ] [@JmillsExpensify / @trjExpensify] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by MelvinBot** [comment]
Looks like something related to `react-navigation` may have been mentioned in this issue discussion.

As a reminder, please make sure that all proposals are not workarounds and that any and all attempt to fix the issue holistically have been made before proceeding with a solution. Proposals to change our `DeprecatedCustomActions.js` files should not be accepted.

Feel free to drop a note in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) with any questions.

**Comment by akshayasalvi** [comment]
@trjExpensify  @aimane-chnaif While this is ready for payout today, is there anything that you want me to do? I am not sure if we would treat this as a bug. This was like a feature request, and we added the category picker in this PR https://github.com/Expensify/App/pull/14532

**Comment by stitesExpensify** [comment]
I agree this is a feature request, not a bug

**Comment by trjExpensify** [comment]
Yep, makes sense. 

Okay, so can I confirm these are the amounts due? 

- @aimane-chnaif - $2,000 PR review + $1,000 #urgency bonus. 
- @akshayasalvi - $2,000 for the fix + $1,000 #urgency bonus.

**Comment by akshayasalvi** [comment]
Yes @trjExpensify I think so :)

**Comment by trjExpensify** [comment]
Cool, thanks. I've sent you both offers!

**Comment by akshayasalvi** [comment]
Accepted @trjExpensify thank you.

**Comment by trjExpensify** [comment]
Excellent, settled up!
